CONTRIBUTING.rst

```diff
@@ -235,7 +235,7 @@ The QA process (``python3 -m nox``) consists of multiple parts:
 
 - linting and checking format(``python3 -m nox --session lint``)
 
-- tests (``python3 -m nox --session tests``)
```

Comment:
Writing `nox -s tests` is significantly easier than `nox -s 'run_tests()'`. This matters because it's a command I type out fairly frequently. It's OK if the low-level commands have difficult to type names, but the high-level ones like qa/lint/tests/docker-qa should remain as short and easy to type as possible.

Reply:
I can remove the braces. The intention was that all versions look the same:
```
- run_tests() -> Run the tests with the default GCC version.
- run_tests(all) -> Run the tests with all GCC versions.
- run_tests(gcc-5) -> Run the test with a specifiv GCC version.
- run_tests(gcc-6) -> Run the test with a specifiv GCC version.
- run_tests(gcc-8) -> Run the test with a specifiv GCC version.
- run_tests(clang-10) -> Run the test with a specifiv GCC version.
```
Oh, there is a typo in the docstring.

Reply:
There is a problem. It seems that the name of a test session which doesn't contain a brace must not be the start of another session. If the session is named `run_tests` it isn't visible, `run_tests(` or `run_tests_` are visible.

Reply:
@latk What about adding a trailing underscore instead of the braces for the default or shall I change back to:
 ```
 - run_tests -> Run the tests with the default GCC version.
 - run_tests_compiler(all) -> Run the tests with all GCC versions.
 - run_tests_compiler(gcc-5) -> Run the test with a specifiv GCC version.
 - run_tests_compiler(gcc-6) -> Run the test with a specifiv GCC version.
 - run_tests_compiler(gcc-8) -> Run the test with a specifiv GCC version.
 - run_tests_compiler(clang-10) -> Run the test with a specifiv GCC version.
```

Reply:
I'm not sure how the trailing underscore style would look, could you give an example? I like the naming in the top pull request comment (e.g. `tests` and `tests_compiler(gcc-5)`).

If there's a problem with session names being a prefix of each other, I'd rather have the more rarely used session have a different prefix.

Reply:
The session name must be unique. The compiler part is parametrized which adds a `(<compiler ID>)` to the name. After your suggestion I've updated the names and teh current output is:
```
Sessions defined in /Users/Shared/Informatik/GitHub/gcovr/noxfile.py:

* qa -> Run the quality tests.
- qa_compiler(gcc-5) -> Run the quality tests.
- qa_compiler(gcc-6) -> Run the quality tests.
- qa_compiler(gcc-8) -> Run the quality tests.
- qa_compiler(gcc-9) -> Run the quality tests.
- qa_compiler(clang-10) -> Run the quality tests.
- qa_compiler(clang-13) -> Run the quality tests.
- lint -> Run the lint (flake8 and black).
- black -> Run black, a code formatter and format checker.
- doc -> Generate the documentation.
- tests -> Run the tests with the default GCC version.
- tests_compiler(all) -> Run the tests with all GCC versions.
- tests_compiler(gcc-5) -> Run the test with a specifiv GCC version.
- tests_compiler(gcc-6) -> Run the test with a specifiv GCC version.
- tests_compiler(gcc-8) -> Run the test with a specifiv GCC version.
- tests_compiler(gcc-9) -> Run the test with a specifiv GCC version.
- tests_compiler(clang-10) -> Run the test with a specifiv GCC version.
- tests_compiler(clang-13) -> Run the test with a specifiv GCC version.
- build_wheel -> Build a wheel.
- check_wheel -> Check the wheel, should not be used directly.
- upload_wheel -> Upload the wheel.
- docker_qa_build -> Build the docker container for the default GCC version.
- docker_qa_build_compiler(all) -> Build the docker containers vor all GCC versions.
- docker_qa_build_compiler(gcc-5) -> Build the docker container for a specific GCC version.
- docker_qa_build_compiler(gcc-6) -> Build the docker container for a specific GCC version.
- docker_qa_build_compiler(gcc-8) -> Build the docker container for a specific GCC version.
- docker_qa_build_compiler(gcc-9) -> Build the docker container for a specific GCC version.
- docker_qa_build_compiler(clang-10) -> Build the docker container for a specific GCC version.
- docker_qa_build_compiler(clang-13) -> Build the docker container for a specific GCC version.
- docker_qa_run -> Run the docker container for the default GCC version.
- docker_qa_run_compiler(all) -> Run the docker container for the all GCC versions.
- docker_qa_run_compiler(gcc-5) -> Run the docker container for a specific GCC version.
- docker_qa_run_compiler(gcc-6) -> Run the docker container for a specific GCC version.
- docker_qa_run_compiler(gcc-8) -> Run the docker container for a specific GCC version.
- docker_qa_run_compiler(gcc-9) -> Run the docker container for a specific GCC version.
- docker_qa_run_compiler(clang-10) -> Run the docker container for a specific GCC version.
- docker_qa_run_compiler(clang-13) -> Run the docker container for a specific GCC version.
- docker_qa -> Build and run the docker container for the default GCC version.
- docker_qa_compiler(all) -> Build and run the docker container for the all GCC versions.
- docker_qa_compiler(gcc-5) -> Build and run the docker container for a specific GCC version.
- docker_qa_compiler(gcc-6) -> Build and run the docker container for a specific GCC version.
- docker_qa_compiler(gcc-8) -> Build and run the docker container for a specific GCC version.
- docker_qa_compiler(gcc-9) -> Build and run the docker container for a specific GCC version.
- docker_qa_compiler(clang-10) -> Build and run the docker container for a specific GCC version.
- docker_qa_compiler(clang-13) -> Build and run the docker container for a specific GCC version.

sessions marked with * are selected, sessions marked with - are skipped.
```

Reply:
I'm happy with the current names. I don't think it's reasonably possible to convince Nox to sort the sessions differently.